.github/workflows/quick.yaml

```
@@ -40,13 +40,21 @@ jobs:
 
   functionality-tests:
 
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
 
     steps:
+      # XXX: A hack to fix Ubuntu runners that were created in April 2025.
+      - name: Fix /etc/hosts
+        run: |
+          cat /etc/hosts
+          cp -p /etc/hosts /tmp/etc-hosts.bak
+          sudo sed --in-place -E 's/^(-e.*)/# \1/' /etc/hosts
+          diff -u /tmp/etc-hosts.bak /etc/hosts || true
+
       - name: Install prerequisite packages
         run: |
           sudo apt-get --quiet=2 update
-          sudo apt-get --quiet=2 install libtool-bin
+          sudo apt-get --quiet=2 install libtool-bin libltdl-dev
```

This test performs a build of Squid and thus requires the ```apt build-dep``` setup from the "build -tests" check. FYI, That environment is required before **any** package build on Debian or Ubuntu.
```suggestion
          sudo apt-get --quiet=2 build-dep squid
          sudo apt-get --quiet=2 install libtool-bin
```


FTR, Other dependencies (like `libtool-bin`) are dependent on which "functionality" are being tested. Right now these "functionality-tests" are very lacking in coverage as shown by all the extra dependencies setup by "build-tests" not being necessary to run the this set of tests.

---

I made this (IMO, out-of-scope) change in branch commit 003293d5 and adjusted PR description accordingly.